core/templates/core/map_page.html

```diff
@@ -0,0 +1,207 @@
+        window.parsedaddresses.push(addressString);
+      }
+
+      // Initialize and add the map
+      function initMap() {
+        var customMapType = new google.maps.StyledMapType([
+            {
+              elementType: 'labels',
+              stylers: [{visibility: 'off'}]
+            },{
+              featureType: "road",
+              stylers: [{visibility: 'on'}]
+            }
+          ], {
+            name: 'Custom Style'
+        });
+
+
+      var customMapTypeId = 'custom_style';
+      // The location of Troy
+      var Troy = {lat: 42.728, lng: -73.691};
+      // The map, centered at Troy
+      var map = new google.maps.Map(
+          document.getElementById('map'), {zoom: 16, center: Troy});
+
+      // Get Lat and Long from addresses
+      var geocoder = new google.maps.Geocoder();
+
+      // create markers per address
+      for(i=0;i<window.parsedaddresses.length;i++){
+        console.log(window.parsedaddresses[i]);
+        console.log(i);
+        geocoder.geocode( { 'address': window.parsedaddresses[i]}, function(results, status) {
+
+        if (status == google.maps.GeocoderStatus.OK) {
+
+
+
+            var latitude = results[0].geometry.location.lat();
+            var longitude = results[0].geometry.location.lng();
+            console.log(results[0].formatted_address);
+
+            var marker = new google.maps.Marker({
+              position: {lat:latitude, lng:longitude},
+              title:"A marker"
+            });
+
+
+
+            var max_value = 0;
+            var max_index = 0;
+            for(j = 0 ;j<window.parsedaddresses.length;j++){
+
+              if(similarity(window.parsedaddresses[j],results[0].formatted_address)> max_value){
+                max_value = similarity(window.parsedaddresses[j],results[0].formatted_address);
+                max_index = j;
+              }
+            }
+
+            var contentString = window.descriptions[max_index];
+
+
+            var infowindow = new google.maps.InfoWindow({
+              content: contentString
+            });
+
+            marker.addListener('click', function() {
+              infowindow.open(map, marker);
+            });
+
+            marker.setMap(map);
+            console.log(contentString);
+
+
+            }
+        });
+
+
+
+
+
+
+
+
+
+
+      // To add the marker to the map, call setMap();
+
+
+    }
+    map.mapTypes.set(customMapTypeId, customMapType);
+    map.setMapTypeId(customMapTypeId);
+}
+    </script>
+    <!--Load the API from the specified URL
+    * The async attribute allows the browser to render the page while the API loads
+    * The key parameter will contain your own API key (which is not needed for this tutorial)
+    * The callback parameter executes the initMap() function
+    -->
+    <script async defer
+    src="https://maps.googleapis.com/maps/api/js?key=AIzaSyD2QD1ngZzv0QjZhylFw0T4_22hzOMjx34&callback=initMap">
```

Comment:
Is this the browser api key? if it is not, embedding it in the frontend will expose the key to the user and could potentially expose us to other people using our key for things we did not intend. We should request the information we need (lat, lng, etc) on the backend and send the information down

Reply:
Yeah in addition to open streets map, I plan on looking into a few other open source map options. Notably, Mapbox, Leaflet, Modest Maps, and Polymaps. Whichever seems to work best. I'll also get on replacing address with lat lng in admin panel.